include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -16,18 +16,299 @@
+        std::ranges::borrowed_iterator_t<_OutRange>>
+    operator()(_ExecutionPolicy&& __exec, _R1&& __r1, _R2&& __r2, _OutRange&& __out_r, _F __binary_op,
+               _Proj1 __proj1 = {}, _Proj2 __proj2 = {}) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+
+        const auto _size = std::ranges::min({std::ranges::size(__r1), std::ranges::size(__r2), std::ranges::size(__out_r)});
+        oneapi::dpl::__internal::__ranges::__pattern_transform(__dispatch_tag, std::forward<_ExecutionPolicy>(__exec),
+            std::ranges::take_view(__r1, _size), std::ranges::take_view(__r2, _size),
+            std::ranges::take_view(__out_r, _size), __binary_op, __proj1, __proj2);
+
+        return {std::ranges::begin(__r1) + _size, std::ranges::begin(__r2) + _size, std::ranges::begin(__out_r) + _size};
+    }
+}; //__transform_fn
+}  //__internal
+
+inline constexpr __internal::__transform_fn transform;
+
+// [alg.find_if]
+
+namespace __internal
+{
+struct __find_if_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _Proj = std::identity,
+             std::indirect_unary_predicate<std::projected<std::ranges::iterator_t<_R>, _Proj>> _Pred>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::ranges::sized_range<_R>
+    auto
+    operator()(_ExecutionPolicy&& __exec, _R&& __r, _Pred __pred, _Proj __proj = {}) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+        return oneapi::dpl::__internal::__ranges::__pattern_find_if(__dispatch_tag,
+            std::forward<_ExecutionPolicy>(__exec), std::forward<_R>(__r), __pred, __proj);
+    }
+}; //__find_if_fn
+}  //__internal
+
+inline constexpr __internal::__find_if_fn find_if;
+
+// [alg.find_if_not]
+
+namespace __internal
+{
+struct __find_if_not_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _Proj = std::identity,
+             std::indirect_unary_predicate<std::projected<std::ranges::iterator_t<_R>, _Proj>>  _Pred>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::ranges::sized_range<_R>
+    auto
+    operator()(_ExecutionPolicy&& __exec, _R&& __r, _Pred __pred, _Proj __proj = {}) const
+    {
+        return oneapi::dpl::ranges::find_if(std::forward<_ExecutionPolicy>(__exec), std::forward<_R>(__r),
+            oneapi::dpl::__internal::__not_pred<oneapi::dpl::__internal::__ref_or_copy<_ExecutionPolicy,
+            _Pred>>(__pred), __proj);
+    }
+}; //__find_if_not_fn
+}  //__internal
+
+inline constexpr __internal::__find_if_not_fn find_if_not;
+
+// [alg.find]
+
+namespace __internal
+{
+struct __find_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _Proj = std::identity,
+             typename _T>
```

Comment:
`_T` precedes `_Proj` in c++20 standard.

*The same incosistency issue was discussed for `search_n`: https://github.com/oneapi-src/oneDPL/pull/1457#discussion_r1687859988.

Reply:
We decide to make it consistent with C++23